code/game/machinery/camera/camera.dm

```diff
@@ -72,9 +67,18 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/camera)
 
 /obj/machinery/camera/Initialize(mapload, obj/structure/camera_assembly/CA)
 	. = ..()
-	for(var/i in network)
-		network -= i
-		network += LOWER_TEXT(i)
+	var/static/list/autonames_in_areas = list()
+
+	// Calculate area code
+	var/area/camera_area = get_area(src)
+	if (istype(camera_area, /area/space))
+		var/turf/connected_wall = get_step(src, dir)
+		camera_area = get_area(connected_wall)
+
+	// Calculate the camera tag
+	if (!c_tag)
+		c_tag = "[format_text(camera_area.name)] #[++autonames_in_areas[camera_area]]"
+	network = camera_area.camera_networks
```

Comment:
shouldn't it be `Copy()`?

Reply:
The list in theory should be readonly so can be a reference, but I'll add vv edit protections to stop admins fucking things up